pulseapi/profiles/views.py

```diff
@@ -66,3 +67,50 @@ def put(self, request, *args, **kwargs):
             pass
 
         return super(UserProfileAPIView, self).put(request, *args, **kwargs)
+
+
+class ProfileCustomFilter(filters.FilterSet):
+    """
+      We add custom filtering to allow you to filter by:
+
+      * Profile type - pass the `?profile_type=` query parameter
+      * Program type - pass the `?program_type=` query parameter
+      * Program year - pass the `?program_year=` query parameter
+    """
+    profile_type = django_filters.CharFilter(
+        name='profile_type__value',
+        lookup_expr='iexact',
+    )
+    program_type = django_filters.CharFilter(
+        name='program_type__value',
+        lookup_expr='iexact',
+    )
+    program_year = django_filters.CharFilter(
+        name='program_year__value',
+        lookup_expr='iexact',
+    )
+
+    class Meta:
+        """
+        Required Meta class
+        """
+        model = UserProfile
+        fields = [
+            'profile_type',
+            'program_type',
+            'program_year',
+        ]
+
+
+class UserProfileAPISearchView(ListCreateAPIView):
+    serializer_class = UserProfileSerializer
+
+    filter_backends = (
+        filters.DjangoFilterBackend,
+        filters.SearchFilter,
+        filters.OrderingFilter,
```

Comment:
do we need this? Are we supporting ordering profiles?

Reply:
more copy paste error from Entries